src/main/java/net/mcreator/workspace/elements/ModElementManager.java

```diff
@@ -89,6 +89,7 @@ public void storeModElement(GeneratableElement element) {
 		}
 
 		cache.put(element.getModElement(), element);
+		workspace.updateModElementTab(element.getModElement());

```

Comment:
It would be wiser to pass GE here and not ME as you have GE at your hand, but you then query it again in updateModElementTab method.

Reply:
But usually `Generator` bases processing on GE instances and here (and in `Workspace`) we work with ME objects.

Reply:
:heavy_check_mark: Fixed

Reply:
Yes, I was also thinking this method may make more sense to be somewhere else. 

For example, translation keys are handled in LocalizationUtils, tags in TagsUtils.

It may make sense to move most of the logic to TabUtils in generator package to follow our conventions

Reply:
Given some of the logic uses workspace instance, how about a class named `TabOrderMap` that would extend a map class currently used?

Reply:
We don't make such structure for neither translation nor tags, so no, just follow the design pattern we used for the existing systems

Reply:
Would this set of methods do it? I'm not sure how to separate them from `Workspace` without own class.

Reply:
Yes, just name of "removeModElement", for example should be something like "removeModElementFromTabs"